docs/book/src/01_intro.md

```diff
@@ -0,0 +1,8 @@
+# Cluster API Operator
```

Comment:
file name could be just `introduction.md` without numbering, but should be fine as a initial base for the book

/lgtm

Reply:
This structure is using https://lib.rs/crates/mdbook-fs-summary - it might be easier to handle summary generation then doing this manually, and less chances that it will leave broken links. That's why it has 01_ in front